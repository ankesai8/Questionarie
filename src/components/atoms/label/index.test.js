import React from "react";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

import { Label } from "./index";

const getChildrenByClass = (tree, classString) => {
  const result = tree.root.find(
    (instance) => instance.props.className === classString
  ).props.children;

  return result;
};

describe("Label", () => {
  it("Label should render properly", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Label  text="Do you have any additional information or comments for us?" />);
    expect(tree).toMatchSnapshot();
  });
});

describe("Label  UI", () => {

  it("should show the correct label text", () => {
    const tree = renderer.create(<Label text="Do you have any additional information or comments for us?" />);
    const text = getChildrenByClass(tree, "label-name");

    expect(text).toBe("Do you have any additional information or comments for us?");

    expect(tree.toJSON()).toMatchSnapshot();
  });

});
