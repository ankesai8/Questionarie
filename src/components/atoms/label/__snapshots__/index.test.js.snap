// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label  UI should show the correct label text 1`] = `
<label
  className="label-name"
>
  Do you have any additional information or comments for us?
</label>
`;

exports[`Label Label should render properly 1`] = `
<label
  className="label-name"
>
  Do you have any additional information or comments for us?
</label>
`;
